sys/net/include/etx_beaconing.h

```diff
@@ -21,27 +21,30 @@
 #define ETX_BEACONING_H_
 #include <stdint.h>
 #include "sixlowpan.h"
-
+/**
+  * makes a function-name in C++ have 'C' linkage so that client C code can link to (i.e use) your function using a 'C' compatible header  file that contains just the declaration of your function
+  */
```

Comment:
You don't need to document `extern "C"` as this is would have to be repeated in every header file in the repo. Besides, I think Doxygen will probably assign this documentation string to the next define instead (`#define ETX_BUF_SIZE`), I have not verified this behaviour however.